mautic/mautic
# Adding down() method to the Migration template

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

- Marking the up() and down() methods to be protected for M6 as they should be in the parent class so they are required.
- Adding down() method
- More examples and suggestions
- Adding a commonly used method for prefixed table name which simplifies the existing migrations but also the new ones.

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Run `bin/console doctrine:migrations:generate` to generate a new migration from this template.
3. Run `bin/console doctrine:migrations:migrate` and make sure it doesn't fail.

#### Other areas of Mautic that may be affected by the change:
1. None

#### List deprecations along with the new alternative:
1. None



[//]: # ( As applicable: )
#### List of areas covered by the unit and/or functional tests:
1. None, as migrations aren't tested.
